src/map/entities/charentity.cpp

```diff
@@ -414,6 +425,11 @@ void CCharEntity::pushPacket(CBasicPacket* packet)
 
     moduleutils::OnPushPacket(this, packet);
 
+    if (CCharEntity::isPacketFiltered(packet))
+    {
+        return;
```

Comment:
Reviewing from bed: if we return here doesn't this leak packet?

Reply:
Could be. that whole pushPacket is basically indecipherable and `packet` here is now unable to be dynamic casted for some unholy reason.

I'll run valgrind

Reply:
Looks like its probably not leaking but i'll remake this PR after everything is converted in #6713